evm/src/test/java/org/hyperledger/besu/evm/operation/PushOperationTest.java

```diff
@@ -34,50 +36,67 @@
 @ExtendWith(MockitoExtension.class)
 public class PushOperationTest {
 
-  private static final byte[] byteCode = new byte[] {0x00, 0x01, 0x02, 0x03};
-  private static final MessageFrame frame =
-      MessageFrame.builder()
-          .worldUpdater(new ToyWorld())
-          .originator(Address.ZERO)
-          .gasPrice(Wei.ONE)
-          .blobGasPrice(Wei.ONE)
-          .blockValues(new ToyBlockValues())
-          .miningBeneficiary(Address.ZERO)
-          .blockHashLookup((l) -> Hash.ZERO)
-          .type(MessageFrame.Type.MESSAGE_CALL)
-          .initialGas(1)
-          .address(Address.ZERO)
-          .contract(Address.ZERO)
-          .inputData(Bytes32.ZERO)
-          .sender(Address.ZERO)
-          .value(Wei.ZERO)
-          .apparentValue(Wei.ZERO)
-          .code(CodeV0.EMPTY_CODE)
-          .completer(messageFrame -> {})
-          .build();
-  ;
+  private final GasCalculator gasCalculator = new PragueGasCalculator();
+  private final CodeFactory codeFactory =
+      new CodeFactory(
+          EvmSpecVersion.PRAGUE.getMaxEofVersion(), EvmSpecVersion.PRAGUE.getMaxInitcodeSize());
+
+  private static final Bytes byteCode = Bytes.wrap(new byte[] {0x00, 0x01, 0x02, 0x03});
+
+  private MessageFrame createMessageFrame(final int pc) {
+    MessageFrame frame =
+        MessageFrame.builder()
+            .worldUpdater(new ToyWorld())
+            .originator(Address.ZERO)
+            .gasPrice(Wei.ONE)
+            .blobGasPrice(Wei.ONE)
+            .blockValues(new ToyBlockValues())
+            .miningBeneficiary(Address.ZERO)
+            .blockHashLookup((l) -> Hash.ZERO)
+            .type(MessageFrame.Type.MESSAGE_CALL)
+            .initialGas(1)
+            .address(Address.ZERO)
+            .contract(Address.ZERO)
+            .inputData(Bytes32.ZERO)
+            .sender(Address.ZERO)
+            .value(Wei.ZERO)
+            .apparentValue(Wei.ZERO)
+            .code(codeFactory.createCode(byteCode))
+            .completer(messageFrame -> {})
+            .build();
+    frame.setPC(pc);
+    return frame;
+  }
 
   @Test
   void unpaddedPushDoesntReachEndCode() {
-    staticOperation(frame, byteCode, 0, byteCode.length - 2);
+    MessageFrame frame = createMessageFrame(0);
+    final Operation operation = new PushOperation(byteCode.size() - 2, gasCalculator);
+    operation.execute(frame, null);
```

Comment:
maybe instead of null, pass a Mock, currently there are no dependencies on EVM but you never know in the future

Reply:
Same for other 3 occurrences in this class

Reply:
did it